src/main.rs

```diff
@@ -151,18 +151,18 @@ fn notmain() -> Result<i32, anyhow::Error> {
     drop(new_linker_script);
 
     // invoke the linker a second time
-    // we need to patch the arguments to use the new linker script
     let mut args = args;
-    // the *hope* is that by placing our temp dir first in the search path (argument list) the
-    // linker will pick our override rather than the original linker script
-    // also add the temp dir to the search path
+    // add the current dir to the linker search path to include all unmodified scripts there
     // HACK `-L` needs to go after `-flavor gnu`; position is currently hardcoded
     args.insert(2, "-L".to_string());
-    args.insert(3, tempdir.path().display().to_string());
-    // we also need to override `_stack_start` to make the stack start below fake RAM
+    args.insert(3, current_dir.to_str().unwrap_or(".").to_string());
```

Comment:
Taking a closer look at the surrounding code, I think we should use `Command::arg` instead of pushing arguments into a `Vec<String>` because `Command::arg` will accept a `Path` directly. That way we can about the string conversion which may be lossy. Could you please do that change as part of this PR?

Reply:
Sure. I was not sure myself about that part. I'll have a look later.